### PR TITLE
tkt-41060: Bug fix for validation in jail plugin

### PR DESCRIPTION
**Pull request by sonicaj**
This commit fixes a bug which prevented valid destination paths to be rejected on validation while updating fstab of a jail.
Ticket: #41060

### CODE REVIEW

File: src/middlewared/middlewared/plugins/jail.py [hunk 1]
```diff
@@ -460,16 +456,17 @@ def fstab(self, jail, options):
...
         index = options.get('index')
 
-        if verrors:
-            raise verrors
-
         if action == 'replace' and index is None:
-            raise ValueError(
-                "Index must not be None when replacing fstab entry"
+            verrors.add(
+                'index',
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/jail.py]: options.index

File: src/middlewared/middlewared/plugins/jail.py [hunk 2]
```diff
@@ -426,21 +426,17 @@ def fstab(self, jail, options):
...
-                    'destination',
-                    'Destination should not include the jail path. '
-                    'e.g "/mnt/iocage/jails/btsync/root/" Path should be specified'
-                    'after /'
-                )
-            elif os.path.exists(dst):
-                if not os.path.isdir(dst):
+            dst = f'{self.get_iocroot()}/jails/{jail}/root'
+            if not dst in destination:
+                destination = f'{dst}{destination}'
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/jail.py]: You're passing the full jail path to iocage fstab, that was not the case before, are you sure thats right?

File: src/middlewared/middlewared/plugins/jail.py [hunk 3]
```diff
@@ -460,16 +456,17 @@ def fstab(self, jail, options):
...
+                'index',
+                'Index must not be None when replacing fstab entry'
             )
 
+        if verrors:
+            raise verrors
+
         _list = iocage.fstab(action, source, destination, fstype, fsoptions,
-                             dump, _pass, index=index, add_path=True)
+                             dump, _pass, index=index)
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/jail.py]: Why removing add_path this time? What does it do?

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/jail.py]: So what add_path does is that if set to true it automatically adds the iocroot to the destination. Now we are allowing users two options. To either specify full path or from the jails root. In both cases this solution would work

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/jail.py]: `            if add_path:
                destination = f"{self.iocroot}/jails/{uuid}/root{destination}"`

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/jail.py]: This snippet is from iocage

File: src/middlewared/middlewared/plugins/jail.py [hunk 4]
```diff
@@ -419,38 +419,34 @@ def fstab(self, jail, options):
...
-                verrors.add(
-                    'destination',
-                    'Destination should not include the jail path. '
-                    'e.g "/mnt/iocage/jails/btsync/root/" Path should be specified'
-                    'after /'
-                )
-            elif os.path.exists(dst):
-                if not os.path.isdir(dst):
+            dst = f'{self.get_iocroot()}/jails/{jail}/root'
+            if not dst in destination:
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/jail.py]: PEP8 error: dst not in destination

**Pull request merged**